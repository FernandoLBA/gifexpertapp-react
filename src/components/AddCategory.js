import React, { useState } from 'react';
import PropTypes from 'prop-types'


export const AddCategory = ( { setCategories } ) => {

     // Creamos un hook
     const [ inputValue, setInputValue ] = useState( '' );

     const handleInputChange = ( e ) => {
          setInputValue( e.target.value );
     };

     const handleSubmit = ( e ) => {
          e.preventDefault();

          if ( inputValue.trim().length > 1 ) {
               // coloca el inputValue primero
               setCategories( categories => [ inputValue, ...categories ] );
               // limpia borra el input
               setInputValue( '' );
          };
     };

     return (
               <form onSubmit={ handleSubmit }>
                    {/* Este párrafo se creó para pasar una prueba por eso no es necesario que se visualice */}
                    <p style={{display:'none' }}>{ inputValue }</p>
                    
                    <input
                         type="text"
                         value={ inputValue }
                         onChange={ handleInputChange }
                         placeholder="Busque aquí..."
                    />
               </form>
     );
};

AddCategory.propTypes = {
     setCategories: PropTypes.func.isRequired
};