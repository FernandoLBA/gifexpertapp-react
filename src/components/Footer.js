import React from 'react'

const Footer = () => {

     const year = new Date().getFullYear();

     return (
          <footer>
               <p>Desarrollada por: Fernando Barrios { year }.</p>
          </footer>
     )
}

export default Footer
