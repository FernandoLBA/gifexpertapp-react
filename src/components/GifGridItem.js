import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ( { title, url } ) => {

     return (
          <div className="card animate__animated animate__zoomIn">
              <img src={ url } alt={ title }></img>
              {/* limita el contenido del texto */}
              {/* <p>{ title.substr( 0, 15 ) }...</p> */}
              <p>{ title }</p>
          </div>
     );
};

export default GifGridItem;

GifGridItem.propTypes = {
     title: PropTypes.string.isRequired,
     url: PropTypes.string.isRequired
};
