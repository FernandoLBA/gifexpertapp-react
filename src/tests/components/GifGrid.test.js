import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');

describe( 'Pruebas al componente <GifGrid />', () => {
     
     const category = 'Goku';

     test( 'Debe mostrar el componente <GifGrid /> correctamente', () => {

          // Establece las propiedades por defecto del state del custom hook useFetchGifs
          useFetchGifs.mockReturnValue({
               data: [],
               loading: true     
          });

          const wrapper = shallow( <GifGrid category={ category }/> );
          expect( wrapper ).toMatchSnapshot();
     } );

     test( 'Debe mostrar items cuando se cargan imágenes con el custom hook useFetchGifs', () => {
          
          // Creamos un array de objetos con datos retornados por el helper getGifs
          const gifs = [{
               id: 'ABC',
               title: 'Cualquier cosa',
               url: 'https://localhost/cualquier/cosa.jpg'
          }];

          // le mandamos el array gifs al state de useFetchGifs
          useFetchGifs.mockReturnValue({
               data: gifs,
               loading: false     
          });

          const wrapper = shallow( <GifGrid category={ category }/> );

          // Esta es la básica
          expect( wrapper ).toMatchSnapshot();

          // Estas son más precisas y específicas
          expect( wrapper.find( 'p' ).exists() ).toBe( false );
          expect( wrapper.find( 'GifGridItem' ).length ).toBe( gifs.length );
     } );
} );
