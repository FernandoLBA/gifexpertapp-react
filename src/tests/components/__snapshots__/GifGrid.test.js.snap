// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGrid /> Debe mostrar el componente <GifGrid /> correctamente 1`] = `
<Fragment>
  <h2>
    Goku
  </h2>
  <p
    className="animate__animated animate__fadeIn"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas al componente <GifGrid /> Debe mostrar items cuando se cargan im√°genes con el custom hook useFetchGifs 1`] = `
<Fragment>
  <h2>
    Goku
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
